test/e2e/cluster_authorized_cidrs_connectivity.go

```diff
@@ -0,0 +1,466 @@
+}, resourceGroup, vmName, command string) (string, error) {
+	subscriptionID, err := tc.SubscriptionID(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	azCreds, err := tc.AzureCredential()
+	if err != nil {
+		return "", err
+	}
+
+	computeClient, err := armcompute.NewVirtualMachinesClient(subscriptionID, azCreds, nil)
+	if err != nil {
+		return "", err
+	}
+
+	runCommandInput := armcompute.RunCommandInput{
+		CommandID: to.Ptr("RunShellScript"),
+		Script: []*string{
+			to.Ptr(command),
+		},
+	}
+
+	poller, err := computeClient.BeginRunCommand(ctx, resourceGroup, vmName, runCommandInput, nil)
+	if err != nil {
+		return "", err
+	}
+
+	result, err := poller.PollUntilDone(ctx, nil)
+	if err != nil {
+		return "", err
+	}
+
+	if len(result.Value) > 0 && result.Value[0].Message != nil {
+		// Azure Run Command returns output in format:
+		// "Enable succeeded: \n[stdout]\n<actual output>\n[stderr]\n<errors>"
+		// We need to extract just the stdout content
+		message := *result.Value[0].Message
+
+		// Find the stdout section
+		stdoutStart := strings.Index(message, "[stdout]\n")
+		if stdoutStart == -1 {
+			// If no stdout marker, return the whole message
+			return message, nil
+		}
+
+		// Skip past the "[stdout]\n" marker
+		stdoutStart += len("[stdout]\n")
+
+		// Find where stderr starts (if present)
+		stderrStart := strings.Index(message[stdoutStart:], "\n[stderr]")
+
+		var output string
+		if stderrStart == -1 {
+			// No stderr marker, take everything after stdout
+			output = message[stdoutStart:]
+		} else {
+			// Take only the stdout section
+			output = message[stdoutStart : stdoutStart+stderrStart]
+		}
+
+		return strings.TrimSpace(output), nil
+	}
+
+	return "", nil
+}
+
+// Helper to test API connectivity with timeout
+func testAPIConnectivity(apiURL string, timeout time.Duration) error {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	// Simple HTTP GET to test connectivity
+	req, err := http.NewRequestWithContext(ctx, "GET", apiURL+"/healthz", nil)
+	if err != nil {
+		return err
+	}
+
+	client := &http.Client{
+		Timeout: timeout,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	return nil
+}
+
+// Helper to generate SSH key pair
+func generateSSHKeyPair() (publicKey string, privateKey string, err error) {
+	// Generate RSA key pair
+	privateKeyData, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return "", "", err
+	}
+
+	// Encode private key to PEM format
+	privateKeyPEM := &pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: x509.MarshalPKCS1PrivateKey(privateKeyData),
+	}
+	privateKeyStr := string(pem.EncodeToMemory(privateKeyPEM))
+
+	// Generate public key in SSH format
+	pub, err := ssh.NewPublicKey(&privateKeyData.PublicKey)
+	if err != nil {
+		return "", "", err
```

Comment:
nit: should we wrap this error with some context like "failed to generate ssh key" or something?